reflect/src/main/java/dagger/reflect/ComponentInvocationHandler.java

```diff
@@ -123,13 +124,22 @@ private ComponentInvocationHandler(Scope scope) {
 
   private static final class ProvisionMethodInvocationHandler implements MethodInvocationHandler {
     private final LinkedBinding<?> binding;
+    private final boolean nullable;
 
-    ProvisionMethodInvocationHandler(LinkedBinding<?> binding) {
+    ProvisionMethodInvocationHandler(LinkedBinding<?> binding, boolean nullable) {
       this.binding = binding;
+      this.nullable = nullable;
     }
 
     @Override
     public @Nullable Object invoke(Object[] args) {
+      if (binding instanceof LinkedProvidesBinding) {
```

Comment:
`@BindsInstance` could also have `@Nullable` can't it?

Reply:
Yeah I think there needs to be something like a `isNullable` on `Binding` until type annotations save us from this nonsense. And then anyone requesting a `Binding` will have to validate that the consumer can handle null.